### PR TITLE
does the quirky thing where a beam with origin/target in a container doesn't cut the map in half because byond is quirky and things inside things are x/y/z=0

### PR DESCRIPTION
**Pull request by silicons**
## Why It's Good For The Game

this is why you have beams going across the map
that's Bad(tm)

## Changelog
:cl:
fix: beams should no longer go across the map and mess everything up if their source or target isn't on a turf.
/:cl:

### CODE REVIEW

File: code/datums/beam.dm [hunk 1]
```diff
@@ -90,6 +92,10 @@
 	return ..()
 
 /datum/beam/proc/Draw()
+	if(!get_turf && (!isturf(src.target.loc) || !isturf(src.origin.loc)))		//byond is quirky! if we aren't getting turf tell it to just Not(tm)
+		return
+	var/atom/origin = get_turf? get_turf(src.origin) : src.origin
+	var/atom/target = get_turf? get_turf(src.target) : src.target
```
Comment (Ghommie) [inline code/datums/beam.dm]: This is probably detrimental and confusing, name these two variables in a different, not omonimous way and early return if one or another are null (which shouldn't be).

File: code/datums/beam.dm [hunk 2]
```diff
@@ -90,6 +92,10 @@
 	return ..()
 
 /datum/beam/proc/Draw()
+	if(!get_turf && (!isturf(src.target.loc) || !isturf(src.origin.loc)))		//byond is quirky! if we aren't getting turf tell it to just Not(tm)
+		return
```
Comment (Ghommie) [inline code/datums/beam.dm]: Remove these.

File: code/datums/beam.dm [hunk 3]
```diff
@@ -15,6 +15,8 @@
 	var/beam_type = /obj/effect/ebeam //must be subtype
 	var/timing_id = null
 	var/recalculating = FALSE
+	/// If either origin or target is in a container, do get_turf instead of just not beaming.
+	var/get_turf = FALSE
```
Comment (Ghommie) [inline code/datums/beam.dm]: This is a pain in the buns and might break a few beams.

**Review by Ghommie — CHANGES_REQUESTED** [review]
TL;DR: just using the turf coordinates for drawing the beam.

**Review by Ghommie — APPROVED** [review]
